src/Microsoft.AspNet.Mvc.TagHelpers/LabelTagHelper.cs

```diff
@@ -45,12 +44,22 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 {
                     output.MergeAttributes(tagBuilder);
 
+                    var childContent = await context.GetChildContentAsync();
+
                     // We check for whitespace to detect scenarios such as:
                     // <label for="Name">
                     // </label>
-                    if (string.IsNullOrWhiteSpace(output.Content))
+                    if (string.IsNullOrEmpty(output.Content))
```

Comment:
Definitely up for discussion. However, i'm leaning towards leaving it as is purely for the ability for `TagHelper`s to reset the `output.Content` if they please (reset = setting to `string.Empty` or `null`). If you're only ever checking `ContentSet` then once it's set there's no going back.